src/datasets/utils/download_manager.py

```diff
@@ -266,7 +271,11 @@ def extract(self, path_or_paths, num_proc=None):
         path_or_paths = NestedDataStructure(path_or_paths)
         extracted_paths = NestedDataStructure(extracted_paths)
         self.extracted_paths.update(dict(zip(path_or_paths.flatten(), extracted_paths.flatten())))
-        return extracted_paths.data
+        return (
+            extracted_paths.data
+            if not isinstance(extracted_paths.data, dict)
+            else defaultdict(str, extracted_paths.data)
+        )
```

Comment:
Why do you need this ?

Reply:
Because the user implementation of the method _split_generators will access all possible keys in the extracted path dict and may certainly access them with __getitem__. If our downloaed hasn't downloaded all possible splits (but only a subset), accessing the missing ones with __getitem__ will raise a `KeyError`.

If a `dafaultdict` is returned instead, trying to access a missing (not downloaded) splits will be silently ignored.

Reply:
Makes sense ! Maybe use a more explicit placeholder that an empty string, in case users experience errors using this ?

Reply:
I've set `"<NOT_DOWNLOADED>"` instead.